Brokerages/Binance/BinanceBrokerage.cs

```diff
@@ -0,0 +1,555 @@
+                    $"{request.Resolution} resolution is not supported, no history returned"));
+                yield break;
+            }
+
+            var period = request.Resolution.ToTimeSpan();
+
+            foreach (var kline in _apiClient.GetHistory(request))
+            {
+                yield return new TradeBar()
+                {
+                    Time = Time.UnixMillisecondTimeStampToDateTime(kline.OpenTime),
+                    Symbol = request.Symbol,
+                    Low = kline.Low,
+                    High = kline.High,
+                    Open = kline.Open,
+                    Close = kline.Close,
+                    Volume = kline.Volume,
+                    Value = kline.Close,
+                    DataType = MarketDataType.TradeBar,
+                    Period = period,
+                    EndTime = Time.UnixMillisecondTimeStampToDateTime(kline.OpenTime + (long)period.TotalMilliseconds)
+                };
+            }
+        }
+
+        /// <summary>
+        /// Event invocator for the OrderFilled event
+        /// </summary>
+        /// <param name="e">The OrderEvent</param>
+        protected void OnOrderSubmit(BinanceOrderSubmitEventArgs e)
+        {
+            var brokerId = e.BrokerId;
+            var order = e.Order;
+            if (CachedOrderIDs.ContainsKey(order.Id))
+            {
+                CachedOrderIDs[order.Id].BrokerId.Clear();
+                CachedOrderIDs[order.Id].BrokerId.Add(brokerId);
+            }
+            else
+            {
+                order.BrokerId.Add(brokerId);
+                CachedOrderIDs.TryAdd(order.Id, order);
+            }
+        }
+
+        /// <summary>
+        /// Wss message handler
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        public override void OnMessage(object sender, WebSocketMessage e)
+        {
+            LastHeartbeatUtcTime = DateTime.UtcNow;
+        }
+
+        private void OnSocketMessage(object sender, WebSocketMessage e, Action<Messages.BaseMessage> handler)
+        {
+            try
+            {
+                if (_streamLocked)
+                {
+                    _messageBuffer.Enqueue(e);
+                    return;
+                }
+            }
+            catch (Exception err)
+            {
+                Log.Error(err);
+            }
+
+            OnMessageImpl(e, handler);
+        }
+
+        /// <summary>
+        /// Force reconnect
+        /// </summary>
+        protected override void Reconnect()
+        {
+            if (WebSocket.IsOpen)
+            {
+                WebSocket?.Close();
+            }
+            base.Reconnect();
+        }
+
+        private void ProcessSubscriptionRequest()
+        {
+            if (_subscriptionsPending) return;
+
+            _lastSubscribeRequestUtcTime = DateTime.UtcNow;
+            _subscriptionsPending = true;
+
+            Task.Run(async () =>
```

Comment:
Think we should replace this approach for a long-running task looking at an event, similar to what we are now doing at `OandaRestApiBase` now and have this method set the event

Reply:
Removed